virttest/qemu_devices/qcontainer.py

```diff
@@ -2904,20 +2904,27 @@ def dimm_device_define_by_params(self, params, name):
         """
         Create pc-dimm device from params.
         """
-        params = params.object_params("dimm")
+        # Ensure get accurate params for dimm device, avoid irrelevant params
+        # be used by mistake.
+        dimm_params = Params()
+        suffix = '_dimm'
+        for key in list(params.keys()):
+            if key.endswith(suffix):
+                new_key = key.rsplit(suffix)[0]
+                dimm_params[new_key] = params[key]
+
         dimm_type = "nvdimm" if params.get("nv_backend") else "pc-dimm"
-        attrs = qdevices.Dimm.__attributes__[:]
-        dev = qdevices.Dimm(params=params.copy_from_keys(attrs),
+        attrs = qdevices.Dimm.__attributes__[dimm_type][:]
+        dimm_uuid = dimm_params.get("uuid")
+        if 'uuid' in attrs and dimm_uuid:
+            if dimm_uuid == "<auto>":
+                dimm_params['uuid'] = uuid.uuid5(uuid.NAMESPACE_OID, name)
+            else:
+                # check if input uuid format is correct
+                uuid.UUID(dimm_uuid)
```

Comment:
The uuid param could be even set with an "invalid" value before your change, so could you please elaborate on the reason of doing so?

Reply:
Seems this line is not used for a param, it just prints the UUID. And here is the background of the previous design: https://github.com/avocado-framework/avocado-vt/pull/2544#discussion_r422486246

Reply:
Yes, it is intentional.  To fail case in this step without start vm.
Because, boot vm with invalid uuid in qemu cli will fail too.

Reply:
I just see what @PaulYuuu shared above: https://github.com/avocado-framework/avocado-vt/pull/2544#discussion_r422663315
Yes, i change this behaviour, to make test fail early. The error message is not same after this change.
1. before, it is error info from qemu process, like: 
```
 Property 'nvdimm.uuid' has invalid value
```

Now, it is exception from uuid:
```
valueError: badly formed hexadecimal UUID string
```
I agree the old design has more clear error msg, while the new design fail case more early.
Actually, both of them are acceptable for me,  any thoughts for this ? @PaulYuuu @luckyh

Reply:
I don't really remember if the previous design was for the qemu negative test, if we ignore negative testing, both of them are acceptable for me.

Reply:
Okay, to cover possible negative testing, change it back to let qemu process raise error for invalid uuid. Thanks!

Reply:
@luckyh @PaulYuuu Could you please help review with the latest change, thanks!